test/e2e.test.ts

```diff
@@ -0,0 +1,139 @@
+  expect(DEBUG_COMMANDS).toBe(false);
+});
+
+type CommandArg = [string, string | string[] | number | boolean];
+const execCommand = (command: string, ...args: CommandArg[]) => {
+  const quote = (val: string) => `"${val}"`;
+  const formattedArgs = args
+    .map(([c, a]) => {
+      // if args is array then quote each elem and separate them with space
+      if (Array.isArray(a)) return `${c} ${a.map(quote).join(' ')}`;
+      // otherwise just quote each elem and separate them with space
+      else return `${c} ${quote(String(a))}`;
+    })
+    .join(' ');
+  const formattedCommand = `${command} ${formattedArgs}`;
+  if (DEBUG_COMMANDS) console.log(`Executing command: ${formattedCommand}`);
+  try {
+    return execSync(`${CLI_EXECUTABLE} ${formattedCommand}`).toString().trim();
+  } catch (e: any) {
+    // rethrow the output of the CLI
+    throw new Error(e.stdout.toString().trim());
+  }
+};
+
+describe('Beacon reader CLI', () => {
+  beforeAll(() => {
+    // By default the built CLI script is not executable and we would need to invoke it using node directly which has a
+    // disadvantage that we are not testing that the shebang works properly.
+    execSync(`chmod +x ${CLI_EXECUTABLE}`);
+  });
+
+  it('shows help', () => {
+    const output = execCommand('--help');
+    expect(output).toMatchSnapshot();
+  });
+
+  it('shows available commands', () => {
+    const output = execCommand('show-available-templates');
+    expect(output).toEqual(['javascript-ethers'].join('\n'));
+  });
+
+  describe('interactive command', () => {
+    const waitOnOutput = (childProcess: ChildProcessWithoutNullStreams) =>
+      new Promise<string>((resolve) =>
+        childProcess.stdout.once('readable', () => {
+          const data = childProcess.stdout.read().toString();
+          resolve(data);
+        })
+      );
+
+    it('works', async () => {
+      // Run the command in a child process so we can pipe stdin to it
+      const childProcess = spawn(CLI_EXECUTABLE);
+
+      // Write the path to the script
+      const tmpDir = createTmpDir();
+      childProcess.stdin.write(tmpDir + '\n');
+      await waitOnOutput(childProcess);
+
+      // Wait a few milliseconds so that the prompt asks for template
+      await new Promise((res) => setTimeout(res, 500));
+
+      // Choose the template to create (accept the default one)
+      childProcess.stdin.write('\n');
+      // Wait a few seconds until the script initializes the project
+      await new Promise((res) => setTimeout(res, 5000));
+      const output = await waitOnOutput(childProcess);
+
+      expect(childProcess.kill()).toBe(true);
+      expect(output).toContain(`Creating a new beacon reader app in ${chalk.green(tmpDir)}`);
+      expect(output).toContain(`Successfully created and initialized a new project.`);
+    }, 10_000);
+  });
+
+  it('parses command arguments', () => {
+    const tmpDir = createTmpDir();
+    const template = 'javascript-ethers';
```

Comment:
I'm wondering why is the name of the template so generic? shouldn't this be `beacon-reader-template` or something like that instead?

Reply:
It's meant to be, because we can later add more templates. E.g. javascript-web3 or typescript-ethers

Also, bear in mind that user will know that this will generate beacon-reader project because he will be using the CLI named `create-beacon-reader-app`